pkgs/development/python-modules/flax/default.nix

```diff
@@ -43,6 +44,7 @@ buildPythonPackage rec {
     jax
     msgpack
     numpy
+    orbax-checkpoint
```

Comment:
I've packaged this for my own needs and just came across this PR which appears to have stalled.

> why is this necessary?

Flax has depended on orbax-checkpoint since 0.6.9. I see that the corresponding tests are disabled, but there is a "runtime dependency check" that fails building Flax without it.
https://github.com/google/flax/blob/48f3a70449ac6dd34b2189fa7934ca9594363a27/CHANGELOG.md?plain=1#L133

Reply:
I guess more precisely my question is: why is this necessary now? We're not changing the version or `src` of flax in any way in this PR. Was it the case that flax was not building successfully prior to this PR?

Reply:
Yes. Using a recent master-branch commit the following fails.
```nix
{ pkgs ? import (builtins.fetchTarball {
    # master Feb 07 2024
    url = "https://github.com/NixOS/nixpkgs/archive/f184fb390100057b21f344aca55ee6d6c8883184.tar.gz";
  }) {}
}:
pkgs.mkShell { buildInputs = [ pkgs.python311 pkgs.python311.pkgs.flax ]; }
```
 I haven't looked into what changed. I assume it's something in the python infrastructure.